DS-Session-6-Assignment/src/question2/GraphImplementation.java

```diff
@@ -0,0 +1,246 @@
+package question2;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.PriorityQueue;
+
+/**
+ * This class implements the methods of the Graph interface
+ * @author Jugal
+ * Dated 08/09/2019
+ */
+public class GraphImplementation implements Graph {
+
+	 private int noOfVertices;
+	    private LinkedList<Vertex> adjacentList[];
+
+	    /**
+	     * Constructor to initialize the data members of the class
+	     * @param noOfVertices is the number of vertices of the graph
+	     */
+	    public GraphImplementation(int noOfVertices) {
+	        if (noOfVertices <= 0) {
+	            System.out.println("Invalid no of vertices:No. Of vertices can't be negative or 0");
+	        }
+	        this.noOfVertices = noOfVertices;
+	        adjacentList = new LinkedList[noOfVertices];
+	        for (int i = 0; i < noOfVertices; ++i)
+	            adjacentList[i] = new LinkedList<Vertex>();
+	    }
+
+	    /** 
+	     * This method adds new edge in Graph
+	     * @param source is the source vertex
+	     * @param destination is the destination vertex
+	     * @param weight is the weight of that Edge
+	     */
+	    public boolean addEdge(int source, int destination, int weight) {
```

Comment:
You should not limit graph nodes to an integer only. They might be in the form of A, B, C .... etc. So Implement the program so we can give any type of nodes.